apps/mobile/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,32 @@
+import { type NativeStackNavigationProp } from "@react-navigation/native-stack";
+import React, { type ReactNode } from "react";
+
+import { RootScreenName } from "~/libs/enums/enums";
+import { useEffect, useNavigation } from "~/libs/hooks/hooks";
+import { type RootNavigationParameterList } from "~/libs/types/types";
+
+type Properties = {
+	children: ReactNode;
+};
+
+const ProtectedRoute: React.FC<Properties> = ({ children }) => {
```

Comment:
There is no need to create a separate component for it. You should just maybe group some existing screens and render them conditionally depending on whether the user is authorized or not.